File: code/insertion_sort.rb
def insertion(notsorted)
  output = File.open("../data/ouput.txt", "w")
  for current in 0..(notsorted.length - 1)
    selected = notsorted[current]
    prev = current - 1
    
    while prev >= 0 && notsorted[prev] > selected
      notsorted[prev+1] = notsorted[prev]
      prev = prev - 1
    end
    notsorted[prev + 1] = selected
    end
  output.puts notsorted
  output.close
  end


input = File.open("../data/input.txt", "r")
input_list = []
while (line = input.gets)
  input_list.push(line)
end



insertion(input_list)

File: code/insertion_sort.py
def insertion(notsorted):
    for current in range(0, (len(notsorted))):
        selected = notsorted[current]
        prev_index = current - 1
        while prev_index >= 0 and notsorted[prev_index] > selected:
            notsorted[prev_index+1] = notsorted[prev_index]
            prev_index = prev_index - 1
        notsorted[prev_index + 1] = selected
    print(notsorted)
    
insertion([9,8,7,6,5,4,3,2,1,9,-2,0,5])
  